### PR TITLE
[SYCL] Update SPIR-V max allowed version to 1.4

### PR DESCRIPTION
**Pull request by MrSidims**
We already use some features from SPIR-V 1.4, for example
partial unrolling.

Signed-off-by: Dmitry Sidorov <dmitry.sidorov@intel.com>

### CODE REVIEW

**Comment by AlexeySachkov** [comment]
[SYCL] Update SPIR-V max ~supported~ allowed version to 1.4

> We already use some features from SPIR-v 1.4

There is no guarantee that it is actually supported by low-level runtime

**Comment by MrSidims** [comment]
Indeed. Well, I've created the PR in the first place to see if there are any issues revealed by pre-commit. And for now I see at least one of them: the translator doesn't know about SPIR-V 1.4.

**Comment by MrSidims** [comment]
@AGindinson please take a look

**Review by AGindinson — APPROVED** [review]

File: clang/lib/Driver/ToolChains/Clang.cpp [hunk 1]
```diff
@@ -8722,7 +8722,7 @@ void SPIRVTranslator::ConstructJob(Compilation &C, const JobAction &JA,
   TranslatorArgs.push_back("-o");
   TranslatorArgs.push_back(Output.getFilename());
   if (JA.isDeviceOffloading(Action::OFK_SYCL)) {
-    TranslatorArgs.push_back("-spirv-max-version=1.3");
+    TranslatorArgs.push_back("-spirv-max-version=1.4");
```
Comment (bader) [inline clang/lib/Driver/ToolChains/Clang.cpp:8725]: @AGindinson, why do we set this option twice?

Comment (MrSidims) [reply clang/lib/Driver/ToolChains/Clang.cpp:8725]: One for translation of the device code (clang.cpp) another one is for possible linking on SPIR-V level (sycl.cpp, but actually there it's a dead code now). At some point I resolve a [comment](https://github.com/intel/llvm/pull/4357#discussion_r695233786) from another PR and we'll set SPIR-V information in a one place.

**Pull request merged**